starter_code/hangman.js

```diff
@@ -1,42 +1,47 @@
 var hangman;
 
-// function Hangman() {
+function Hangman() {
+  this.words = ["hello","please","thanks"];
+  this.secretWord = "";
+  this.letters = [];
+  this.guessedLetters = "";
+  this.errorsLeft = 10;
+}
 
-// }
+Hangman.prototype.getWord = function () {
+  return this.words[Math.floor(Math.random()*this.words.length)];
 
-// Hangman.prototype.getWord = function () {
-
-// };
-
-// Hangman.prototype.checkIfLetter = function (keyCode) {
-
-// };
+};
 
-// Hangman.prototype.checkClickedLetters = function (key) {
+Hangman.prototype.checkIfLetter = function (keyCode) {
+    return ((keyCode>64 && keyCode<91) ? true  : false );
+};
 
-// };
+Hangman.prototype.checkClickedLetters = function (key) {
+  return !this.letters.includes(key);
+};
 
-// Hangman.prototype.addCorrectLetter = function (i) {
+Hangman.prototype.addCorrectLetter = function (i) {
+//   return (this.clickedLetters) ? this.guessedLetter += this.secretWord[i].toUpperCase() :
```

Comment:
Basically what you have to do here is to assign the value of the letter that is equal to the index "i" in the secretword, so you can use something like this : 
```js
  this.guessedLetter += this.secretWord[i].toUpperCase();
  return this.guessedLetter;
```